### ISSUE TITLE
The "extra_groups" attribute in <UnixGroup .../> not required?

### ISSUE THREAD

**Issue by dikim33** [weiss/bcfg2-accounts#2]
Hi,

I found that the "extra_groups" attribute in <UnixGroup ... /> is not necessary and it may be a trouble because when I ran "bcfg2-repo-validate" with the "extra_groups" groups in <UnixGroup ... />, I got the following error.

[root@leopard share]# bcfg2-repo-validate
ERROR: /var/lib/bcfg2/Properties/accounts.xml fails to verify:
  /var/lib/bcfg2/Properties/accounts.xml:80: element UnixGroup: Schemas validity error : Element 'UnixGroup', attribute 'extra_groups': The attribute 'extra_groups' is not allowed.
  /var/lib/bcfg2/Properties/accounts.xml fails to validate
Serverless plugins encountered errors, skipping server plugins
1 errors
0 warnings
[root@leopard share]#

But I did not have any errors of interpreting the accounts.xml data in the client side when I ran "bcfg2 -qvn" with this attribute.
The bottom line is that it generates all the groups properly whether the "extra_groups" attribute is there or not.

I think that this is not really an issue but I had to follow the instruction of bcfg2-repo-validate whether it fails with "bcfg2 -qv" or not.

**Comment by weiss** [comment]
According to the error message, you specified `extra_groups` as a `UnixGroup` attribute, this is indeed not allowed.  `extra_groups` is an optional `UnixUser` attribute which can be used to specify a (space-separated) list of additional groups this _user_ should be part of.
